Client/Frontend/Browser/PlaylistHelper.swift

```diff
@@ -63,98 +72,137 @@ class PlaylistHelper: NSObject, TabContentScript {
     }
     
     func userContentController(_ userContentController: WKUserContentController, didReceiveScriptMessage message: WKScriptMessage) {
+        guard let item = PlaylistInfo.from(message: message), !item.src.isEmpty else {
+            DispatchQueue.main.async {
+                self.delegate?.openInPlayListActivity(info: nil)
+                self.delegate?.addToPlayListActivity(info: nil, itemDetected: false)
+            }
+            return
+        }
+        
         PlaylistHelper.queue.async { [weak self] in
             guard let self = self else { return }
-            
-            guard let item = DispatchQueue.main.sync(execute: { PlaylistInfo.from(message: message) }),
-                  !item.src.isEmpty else {
-                ensureMainThread {
-                    self.delegate?.openInPlayListActivity(info: nil)
-                    self.delegate?.addToPlayListActivity(info: nil, itemDetected: false)
-                }
-                return
-            }
-            
+
             if item.duration <= 0.0 && !item.detected || item.src.isEmpty || item.src.hasPrefix("data:") || item.src.hasPrefix("blob:") {
-                ensureMainThread {
+                DispatchQueue.main.async {
```

Comment:
Why did you wanna specifically change `ensureMainThread` ? Can it be synchronous ?

Reply:
Because `ensureMainThread` can be either synchronous or async. I want it to always run async and not synchronous because WebKit can potentially freeze and wait for the JS callback to finish. I don't want that at all :)